### ISSUE TITLE
BUG: GPU logging fails on NVIDIA MULTI-INSTANCE GPU

### ISSUE THREAD

**Issue by KaleabTessera** [neptune-ai/neptune-client#790]
### Describe the bug
Gpu logging fails for NVIDIA MULTI-INSTANCE GPU (https://www.nvidia.com/en-us/technologies/multi-instance-gpu/) with an a100, running in a docker container (image - https://hub.docker.com/layers/ktessera/mava/tf-core-latest/images/sha256-9c3a0ab55e4f9f42aee8c934ef7ca3db0c5fc808f193ac72ff70bee70106b88d?context=repo, based on https://catalog.ngc.nvidia.com/orgs/nvidia/containers/cuda/tags - tag:11.4.2-devel-ubuntu20.04).
### Reproduction
Run anything on gpu and attempt to log to neptune.
### Expected behavior
To see gpu memory etc. 
### Traceback
```
2021/12/21 13:01:09 | Exception in thread
2021/12/21 13:01:09 | Thread-2
2021/12/21 13:01:09 | :
2021/12/21 13:01:09 | Traceback (most recent call last):
2021/12/21 13:01:09 | File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | self.run()
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/new/internal/threading/daemon.py", line 53, in run
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | self.work()
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/new/internal/hardware/hardware_metric_reporting_job.py", line 120, in work
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | metric_reports = self._metric_reporter.report(time.time())
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py", line 32, in report
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | return [
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py", line 37, in <listcomp>
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | for x in [
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py", line 38, in <listcomp>
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | self.__metric_value_for_gauge(gauge, timestamp)
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py", line 48, in __metric_value_for_gauge
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | value = gauge.value()
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | File "/usr/local/lib/python3.8/dist-packages/neptune/internal/hardware/gauges/gpu.py", line 48, in value
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | return self.__gpu_monitor.get_card_used_memory_in_bytes(
2021/12/21 13:01:09 |  
2021/12/21 13:01:09 | TypeError
2021/12/21 13:01:09 | :
2021/12/21 13:01:09 | unsupported operand type(s) for /: 'NoneType' and 'float'
2021/12/21 13:01:09
```

### Environment

Tried multiple versions of the neptune client, including 0.14.0. 

### Additional context
Processes not using gpu, can log cpu and memory usage with no issues.

**Comment by Blaizzy** [comment]
Hey @KaleabTessera,

Prince Canuma here, a Data Scientist at Neptune.ai

Can you send me a minimal code that reproduces the error?

**Comment by Blaizzy** [comment]
Does this error break your process or just logs an exception and while logging your metadata(metrics, files, and so on) normally?

**Comment by KaleabTessera** [comment]
Hi @Blaizzy , thanks for looking at this. 

So the error doesn't break our process, it just fails to log gpu usage and logs an exception. I found the full exception:
```
tensorflow 2022-01-06T17:42:11.34: [variable_server/0] U[trainer/0] I0106 11:07:21.478492 140125155747648 node.py:62] Reverb client connecting to: localhost:22151
tensorflow 2022-01-06T17:42:11.34: [trainer/0] https://app.neptune.ai/InstaDeep/Mava/e/MAV-955
tensorflow 2022-01-06T17:42:11.34: [trainer/0] /mava/lib/python3.8/site-packages/neptune/new/internal/utils/git.py:35: UserWarning: GitPython could not be initialized
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   warnings.warn("GitPython could not be initialized")
tensorflow 2022-01-06T17:42:11.34: [trainer/0] /mava/lib/python3.8/site-packages/neptune/new/internal/utils/git.py:35: UserWarning: GitPython could not be initialized
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   warnings.warn("GitPython could not be initialized")
tensorflow 2022-01-06T17:42:11.34: [trainer/0] /mava/lib/python3.8/site-packages/neptune/new/internal/utils/git.py:35: UserWarning: GitPython could not be initialized
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   warnings.warn("GitPython could not be initialized")
tensorflow 2022-01-06T17:42:11.34: [trainer/0] W0106 11:07:22.487560 140125155747648 gpu_monitor.py:87] Info (NVML): Insufficient Permissions. GPU usage metrics may not be reported. For more information, see https://docs-legacy.neptune.ai/logging-and-managing-experiment-results/logging-experiment-data.html#hardware-consumption 
tensorflow 2022-01-06T17:42:11.34: [trainer/0] Exception in thread Thread-2:
tensorflow 2022-01-06T17:42:11.34: [trainer/0] Traceback (most recent call last):
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/usr/lib/python3.8/threading.py";, line 932, in _bootstrap_inner
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     self.run()
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/new/internal/threading/daemon.py";, line 53, in run
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     self.work()
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/new/internal/hardware/hardware_metric_reporting_job.py";, line 120, in work
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     metric_reports = self._metric_reporter.report(time.time())
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py";, line 32, in report
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     return [
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py";, line 37, in <listcomp>
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     for x in [
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py";, line 38, in <listcomp>
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     self.__metric_value_for_gauge(gauge, timestamp)
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/internal/hardware/metrics/reports/metric_reporter.py";, line 48, in __metric_value_for_gauge
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     value = gauge.value()
tensorflow 2022-01-06T17:42:11.34: [trainer/0]   File "/mava/lib/python3.8/site-packages/neptune/internal/hardware/gauges/gpu.py";, line 48, in value
tensorflow 2022-01-06T17:42:11.34: [trainer/0]     return self.__gpu_monitor.get_card_used_memory_in_bytes(
tensorflow 2022-01-06T17:42:11.34: [trainer/0] TypeError: unsupported operand type(s) for /: 'NoneType' and 'float'
```

Unfortunately, for now it is difficult to provide a minimal code example that reproduces this error as it appears to only happen when running on certain types of gpus. I will confirm this and post a reply. 

I will also try the version of the neptune client (0.14.0 -> 0.14.1)  as I see some changes were made related to gpu logging.

**Comment by KaleabTessera** [comment]
Same issue with `0.14.1`.

**Comment by Blaizzy** [comment]
Hey @KaleabTessera 

Thank you very much for the detailed traceback.

I noticed that the issue you are having is very common on NVIDIA GPUs which is **NVML Error: Insufficient Permissions**.
Your traceback:
`
tensorflow 2022-01-06T17:42:11.34: [trainer/0] W0106 11:07:22.487560 140125155747648 gpu_monitor.py:87] Info (NVML): Insufficient Permissions. GPU usage metrics may not be reported...
`

I found this closed issue that helps you solves the problem in the NVIDIA GitHub repo: https://github.com/NVIDIA/nvidia-docker/issues/1547

Basically, what's happening is:
_"Any non-root user not belonging to the video group will encounter the NVML: Insufficient Permissions error when using NVIDIA utilities such as nvidia-smi . Thus, the proper fix for this is to add your user to the video group then relogin or **access it as a root user**(according to the NVDIA GH repo solution above 👆) then relogin"_ .
Source: https://www.reddit.com/r/openSUSE/comments/gt9kiy/nvidia_on_tumbleweed_a_fix_for_the_nvml/#:~:text=The%20Solution,the%20video%20group%20then%20relogin.

**Comment by Blaizzy** [comment]
@KaleabTessera please try it out and let me know if this solves your problem 🤓 !

**Comment by Blaizzy** [comment]
Hey @KaleabTessera,

Any updates?

**Comment by Blaizzy** [comment]
Hey @KaleabTessera,

How are you doing?

Just checking in on your issue, did you manage to solve it?

**Comment by KaleabTessera** [comment]
Hi @Blaizzy, yeah sorry for the late reply. 

We can close this. I realized `nvidia-smi` wasn't working through the container so this is not a neptune issue.
